vc-got.el

```diff
@@ -432,6 +431,22 @@ DIR-OR-FILE."
   "Register FILES, passing `vc-register-switches' to the backend command."
   (vc-got--add files))
 
+(defun vc-got-unregister (file)
+  "Unregister given FILE, i.e. remove file if it was versioned
+file or revert it if it was added but not committed."
+  (pcase ((status (cdr (vc-got--parse-status (vc-got--status file)))))
+    ('unregistered nil) ;; no need for action
+    ((or 'added 'missing) (vc-got--revert file))
+    (default (vc-got--remove file))))
+
+(defun vc-got--remove (file &optional force keep-local)
+  "Internal helper to removing FILE from got."
+  (vc-got--call "remove" (if force "-f" "") (if keep-local "-k" "")))
```

Comment:
This doesn't seem right.  You're not passing `file` to `vc-got--call`, and those two `if`s are wrong.  If force is nil, an empty string is added after "remove", and this will confuse got (which will try to remove the `""` file).

To avoid duplicating what was done in `vc-got--log`, I pushed a commit to change how `vc-got--call` handles arguments.  Now you can safely pass `nil`s and lists (like `'("-l" "2")`) to `vc-got--call`.

So, you can rewrite that as `(vc-got--call "remove" (when force "-f") (when keep-local "-k") file)`

Reply:
yeah, I pushed this draft to let you know what I'm working on so we won't do duplicate work. I didn't do any tests on them at that point.

Reply:
No problem :)  Thanks for the efforts!

I think this is good to merge, is it?